public/js/monitor.js

```diff
@@ -57,61 +57,82 @@ function isValidEmail(val) {
   return re.test(String(val).toLowerCase());
 }
 
-function addClass(selector, className){
+function addClass(selector, className) {
   const el = document.querySelector(selector);
-  if(el){
+  if (el) {
     el.classList.add(className);
   }
 }
 
-function removeClass(selector, className){
+function removeClass(selector, className) {
   const el = document.querySelector(selector);
-  if(el){
+  if (el) {
     el.classList.remove(className);
   }
 }
 
-function removeInvalidMessage(){
+function removeInvalidMessage() {
   removeClass("#invalid-email-message", "show");
   removeClass(".input-group","invalid");
 }
 
-function addInvalidMessage(){
+function addInvalidMessage() {
   addClass("#invalid-email-message", "show");
   addClass(".input-group", "invalid");
 }
 
 function enableBtnIfEmailValid(e) {
+  removeInvalidMessage();
   const emailBtn = document.getElementById("submit-email");
   if (isValidEmail(e.target.value)) {
       emailBtn.disabled = false;
       removeInvalidMessage();
   } else {
       emailBtn.disabled = true;
+      if(e.code === "Enter") {
+        addInvalidMessage();
+      }
   }
 }
 
 function enableBtnIfInputEmpty(e) {
   const emailBtn = document.getElementById("submit-email");
-  if(e.target.value === ""){
+  if (e.target.value === "") {
       emailBtn.disabled = false;
       removeInvalidMessage();
   } else {
       enableBtnIfEmailValid(e);
   }
 }
 
-function showFalseDoor(){
+function hideFalseDoor() {
+  addClass("#false-door", "hidden");
+  document.body.style.overflow = "scroll";
+}
+
+function showFalseDoor() {
   handleEvents("SignUp");
   removeClass("#false-door", "hidden");
-  document.getElementById("close-false-door").onclick = function (){
-    addClass("#false-door", "hidden");
-  };
+  document.body.style.overflow = "hidden";
+  const falseDoor = document.getElementById("false-door");
+  window.addEventListener("keydown", function listener(e) {
+    if (e.code !== "Enter" && e.code !== "Escape") {
+      return;
+    }
+    hideFalseDoor();
+    window.removeEventListener("keydown", listener);
+  });
+  falseDoor.addEventListener("click", function listener (e) {
```

Comment:
I'll be honest and tell you that I don't exactly know how this works (but trust that you and @nhnt11 know what's going on).
But my crude understanding is that you have a function named `listener()` here (and on L118) which does some stuff. I'm not sure how the variable hoisting works in this scenario since my guess is that you're overwriting the hoisted `listener()` function declared earlier. I'd have to try adding a bunch of `console.log()` statements throughout to try and verify the exact hoisting rules in this exact case.
Or, just rename the listener functions to be more unique ("windowKeydownListener" and "falseDoorClickListener") and cross our fingers.

Reply:
yes, good point, let's name these two listeners differently.